cloud-sql/mysql/Program.cs

```diff
@@ -65,6 +73,142 @@ private static void ReadAppSettings()
 
     static class ProgramExtensions
     {
+        public static DbConnection OpenWithRetry(this DbConnection connection)
+        {
+            // [START cloud_sql_mysql_dotnet_ado_backoff]
+            connection = Policy
+                .HandleResult<DbConnection>(conn => conn.State != ConnectionState.Open)
+                .WaitAndRetry(new[]
+                {
+                TimeSpan.FromSeconds(1),
+                TimeSpan.FromSeconds(2),
+                TimeSpan.FromSeconds(5)
+                }, (result, timeSpan, retryCount, context) =>
+                {
+                    // Log any warnings here.

```

Comment:
OK, but I think now, after the third retry, if the connection has not been opened, it will be returned nonetheles, closed. And then failure will probably happen when a command is ran. Maybe here, if the retryCount is 3, and result contains an exception, just throw it?

Reply:
Added a check that throws if retry is 3 and result contains an exception.